### DISCUSSION TITLE
Suggestion: Secure PIN Entry

### DISCUSSION THREAD

**Discussion by worldjoe** [pannal/plex-for-kodi#126] [Ideas]
On my NVidia Shield my kids can see me type in my PIN. I figured this was controlled by plex, but I see [code](https://github.com/pannal/plex-for-kodi/blob/9fa6e376f07a6dfc1491dc2b512ae25f107421f5/lib/windows/signin.py#L92) that makes it look like the pin entry is controlled by the plugin. [Plex forums](https://forums.plex.tv/t/hide-pin-entry-so-not-everybody-in-the-room-can-see-it-mod-enter-pin-via-remote-s-keypad/318833) have been discussing this problem since 2016. Honestly it seems like a security bug not a feature request.
The obvious fix is to switch to an Amazon Fire like pin entry system seen here: https://sm.pcmag.com/pcmag_ap/photo/p/parental-c/parental-control-settings_day5.jpg

**Comment by worldjoe** [comment]
I'm a capable programmer, so I guess what I'm asking is, does plex-for-kodi control the input mechanism? If so, I will try to tackle this. With new (to me) code bases, it's not super clear what might be dead code, or what might be annotations and not core code that drives the input of the pin code. Thanks.

**Comment by pannal** [comment]
Yeah we control the pin entry. Feel free to work on this, I have my hands full with 0.7.9 right now.

**Reply by pannal** [reply]
It's very old code, but still in use. The project has relatively few dead code areas.

**Comment by pannal** [comment]
Did you manage to take a look at this?

**Reply by pannal** [reply]
@worldjoe bump? :)
